include/boost/numeric/ublas/tensor/functions.hpp

```diff
@@ -14,87 +14,90 @@
 #ifndef _BOOST_UBLAS_TENSOR_FUNCTIONS_HPP_
 #define _BOOST_UBLAS_TENSOR_FUNCTIONS_HPP_
 
-
-#include <stdexcept>
-#include <vector>
-#include <algorithm>
-#include <numeric>
-
 #include <boost/numeric/ublas/tensor/multiplication.hpp>
-#include <boost/numeric/ublas/tensor/algorithms.hpp>
-#include <boost/numeric/ublas/tensor/expression.hpp>
-#include <boost/numeric/ublas/tensor/expression_evaluation.hpp>
-#include <boost/numeric/ublas/tensor/type_traits.hpp>
-#include <boost/numeric/ublas/tensor/fixed_rank_extents.hpp>
-#include <boost/numeric/ublas/tensor/static_extents.hpp>
-#include <boost/numeric/ublas/tensor/dynamic_extents.hpp>
-#include <boost/numeric/ublas/tensor/strides.hpp>
-
-namespace boost::numeric::ublas{
-    
-    template<typename T, typename F>
-    struct dynamic_tensor;
-    
-    template<typename T, typename E, typename F>
-    struct static_tensor;
-    
-    template<typename T, std::size_t R, typename F>
-    struct fixed_rank_tensor;
-
-} // namespace boost::numeric::ublas
-
+#include <boost/numeric/ublas/tensor/tensor.hpp>
```

Comment:
I will change it to include `tensor_engine.hpp`, if we don't include it and forward declare it then we won't be able to create the result type because it needs to know how to construct the object. We cannot forward declare a class. There is another way which is including it after defining `tensor_engine` in the same header file or include it after the including `tensor_enging.hpp`.

Reply:
You are right my mistake - I confused `functions.hpp` with `multiplication.hpp`. I should have named those files differently.
I would not like to have ordering dependencies. So it is okay like that.